idd/Energy+.idd.in

```diff
@@ -16134,6 +16134,27 @@ SurfaceProperty:HeatTransferAlgorithm:Construction,
        \type object-list
        \object-list ConstructionNames
 
+SurfaceProperty:HeatBalanceSourceTerm,
+       \memo Allows additional heat source term of the surface boundary condition in the surface heat balance calculation
+       \memo For Advanced/Research Usage
```

Comment:
I do not like this.  It's either suitable for use by all users or it doesn't go in the program.  This line can go away.

Reply:
Suggest that the memo read "Allows an additional heat source term to be added to the inside or outside surface boundary."

Do you allow a source to be added to both the inside and outside of the same surface?  If not, then this should be explained in the memo.  If yes, then this input structure has validation problems, because there could be more than one object with the same name. Does your code check to make sure that there is not more than one object setting, say, an inside source on the same surface?  I'm not sure about the other surfaceproperty objects, but I think the surface names must be unique across all SurfaceProperty:ConvectionCoefficients objects.

An alternate input design would be
```
SurfaceProperty:HeatBalanceSourceTerm,
 A1 , \field Surface Name
A2 ; \field Outside Surface Heat Source Term Schedule Name
A3 ; \field Inside Surface Heat Source Term Schedule Name
```
This would allow getinput to check for unique surface names within this class. The schedule name fields would then be optional instead of required if this approach is taken.  If blank, then there is no source on that side.

Also-do you check and throw and error if this is applied to a surface that has a special boundary condition such as OtherSideCoefficients or OtherSideConditions? 

Does it work with an interzone surface (outside boundary = surface or zone)?

Reply:
Or better might be "Outside Face Heat Source Term . . " and "Inside Face . . . " to match the terminology.

Also - is there a new output variable for these terms?  If not, we need them, suggested names:
Surface Inside Face Heat Source Gain Rate per Area [W/m2]
Surface Inside Face Heat Source Gain Rate [W]
similar for Surface Outside Face . ..

Reply:
@Myoldmopar @mjwitte 
The feature is supposed to allow either or both sides of a surface. And it is supposed to work with any surfaces in all conditions. So we
- Updated the IDD
- Modified the docs
- Checked for unique surface name
- Threw errors if OtherSideCoefficients